test/AsmResolver.Workspaces.DotNet.Tests/AssemblyTraversalTest.cs

```diff
@@ -0,0 +1,268 @@
+             var fileReferences = GetAllMembers<FileReference>(_fixture.AllAssemblies, TableIndex.File);
+            TraverseObjects(fileReferences);*/
+            var assembly = new AssemblyDefinition("Assembly", new Version(1, 0, 0, 0));
+            var module = new ModuleDefinition("Module");
+
+            var file1 = new FileReference("SomeName1", FileAttributes.ContainsMetadata);
+            var file2 = new FileReference("SomeName2", FileAttributes.ContainsMetadata);
+
+            assembly.Modules.Add(module);
+            module.FileReferences.Add(file1);
+            module.FileReferences.Add(file2);
+
+            var workspace = new DotNetWorkspace();
+            var context = new AnalysisContext(workspace);
+
+            workspace.Assemblies.Add(assembly);
+            context.SchedulaForAnalysis(assembly);
+
+            workspace.Analyze(context);
+
+            Assert.True(context.TraversedObjects.Contains(file1));
+            Assert.True(context.TraversedObjects.Contains(file2));
+        }
+
+        [Fact]
+        public void ExportedTypeCheck()
+        {
+            var exportedTypes = GetAllMembers<ExportedType>(_fixture.AllAssemblies, TableIndex.ExportedType);
+            TraverseObjects(exportedTypes);
+        }
+
+        [Fact]
+        public void ManifestResourceCheck()
+        {
+            var manifestResources = GetAllMembers<ManifestResource>(_fixture.AllAssemblies, TableIndex.ManifestResource);
+            TraverseObjects(manifestResources);
+        }
+
+        [Fact]
+        public void GenericParamCheck()
+        {
+            var genericParameters = GetAllMembers<GenericParameter>(_fixture.AllAssemblies, TableIndex.GenericParam);
+            TraverseObjects(genericParameters);
+        }
+
+        [Fact]
+        public void MethodSpecCheck()
+        {
+            var methodSpecifications = GetAllMembers<MethodSpecification>(_fixture.AllAssemblies, TableIndex.MethodSpec);
+            TraverseObjects(methodSpecifications);
+        }
+
+        [Fact]
+        public void GenericParamConstraintCheck()
+        {
+            var genericParameterConstraints =
+                GetAllMembers<GenericParameterConstraint>(_fixture.AllAssemblies, TableIndex.GenericParamConstraint);
+            TraverseObjects(genericParameterConstraints);
+        }
+
+        private void TraverseObjects(IEnumerable<object> members)
+        {
+            Assert.NotEmpty(members);
+
+            var workspace = new DotNetWorkspace();
+            var context = new AnalysisContext(workspace);
+
+            foreach (var assembly in _fixture.AllAssemblies)
+            {
+                workspace.Assemblies.Add(assembly);
+                context.SchedulaForAnalysis(assembly);
+            }
+
+            workspace.Analyze(context);
+
+            Assert.All(members, member
+                => Assert.True(context.TraversedObjects.Contains(member)));
```

Comment:
Use `Assert.Contains(list, element)` instead of `Assert.True(list.Contains(element))`

Reply:
It's not allowed since TraversedObjects is ISet<object>